service/worker/scheduler/workflow.go

```diff
@@ -861,6 +875,14 @@ func (s *scheduler) processUpdate(req *schedspb.FullUpdateRequest) {
 	s.ensureFields()
 	s.compileSpec()
 
+	if s.hasMinVersion(UpdateFromPrevious) {
```

Comment:
You have version checks in many different functions. We should extract them into a separate strategy. We can do strategy per version or just encapsulate all the version checks inside one small file. Then from here it would be just `s.strategy.UpdateLastProcessedTime`.

Reply:
Yes and no.. in general I might agree. But the main thing is the diffs: because of the requirements for workflow determinism, it has to be extremely obvious from looking at a diff that introduces a version check that the "before" path is identical to the pre-diff path. If we moved the actual different code into another file then you'd have to go jump to that file and think harder to verify that condition. Using inline checks clutters up the code but makes it possible to verify that with only local information in an almost mechanical way.